s3/container.go

```diff
@@ -43,11 +44,24 @@ func (c *container) PreSignRequest(ctx context.Context, clientMethod stow.Client
 			contentMD5 = aws.String(params.ContentMD5)
 		}
 
-		req, _ = c.client.PutObjectRequest(&s3.PutObjectInput{
+		params := &s3.PutObjectInput{
 			Bucket:     aws.String(c.name),
 			Key:        aws.String(id),
 			ContentMD5: contentMD5,
-		})
+		}
+		if bucketEncrypted, sseAlgortihm, encryptionKey := getKmsMasterKeyId(c.client, c.name); bucketEncrypted {
+			switch sseAlgortihm {
+			case s3.ServerSideEncryptionAes256:
+				params.ServerSideEncryption = aws.String(sseAlgortihm)
+			case s3.ServerSideEncryptionAwsKms:
+				params.ServerSideEncryption = aws.String(sseAlgortihm)
+				if encryptionKey != "" {
+					params.SSEKMSKeyId = aws.String(encryptionKey)
```

Comment:
I believe this header only needs to be set on PUT-adjacent operations (i.e. not GET/DELETE). Compare to e.g. https://github.com/search?q=repo%3Atreeverse%2FlakeFS%20SSEKMSKeyId&type=code